libs/clients/social-insurance-administration/src/lib/socialInsuranceAdministrationClient.module.ts

```diff
@@ -0,0 +1,72 @@
+import { DynamicModule, Module } from '@nestjs/common'
+import fetch from 'isomorphic-fetch'
+
+import { logger } from '@island.is/logging'
+import { isRunningOnEnvironment } from '@island.is/shared/utils'
+
+import {
+  Configuration,
+  GetApplicantInfoApi,
+  SendApplicationApi,
+  GetIsApplicantEligibleApi,
+  GetCurrenciesApi,
+} from '../../gen/fetch'
+import { createWrappedFetchWithLogging } from './utils'
+import { SocialInsuranceAdministrationClientService } from './socialInsuranceAdministrationClient.service'
+
+const isRunningOnProduction = isRunningOnEnvironment('production')
+
+export interface SocialInsuranceAdministrationModuleConfig {
+  apiKey: string
+  xRoadPath: string
+  xRoadClient: string
+}
+
+@Module({
+  providers: [SocialInsuranceAdministrationClientService],
+  exports: [SocialInsuranceAdministrationClientService],
+})
+export class SocialInsuranceAdministrationClientModule {
+  static register(
```

Comment:
Please use the [`ConfigModule`](https://github.com/island-is/island.is/tree/main/libs/nest/config) and inject to the factoryFunction in the provider.